Snakefile

```diff
@@ -13,23 +13,27 @@ SEP = '/'
 wildcard_constraints:
     params="params-\w+"
 
-config, datasets, out_dir, algorithm_params, algorithm_directed, pca_params, hac_params = process_config(config)
+# Elsewhere we import this as config, but in the Snakefile, the variable config is already populated
+# with the parsed config.yaml. This is done by Snakemake, which magically pipes config into this file
+# without declaration!
+_config.init_global(config)
 
-# TODO consider the best way to pass global configuration information to the run functions
-SINGULARITY = "singularity" in config and config["singularity"]
-if SINGULARITY:
-    print('Running Singularity containers')
-else:
-    print('Running Docker containers')
+out_dir = _config.config.out_dir
```

Comment:
I agree that using `_config.config` is clunky, but I don't have a good solution for this. For example, in `containers.py`, the following
```
import src.config as config
hash_length = config.config.hash_length
```
correctly grabs the instantiated `config` object and the configured hash length, but if I instead do
```
from src.config import config
hash_length = config.hash_length
```
I get an uninstantiated config object (of type `None`).

Furthermore, even a reassignment seems to break the instantiated object because the following also does not work:
```
import src.config as _config
my_config = _config.config
hash_length = my_config.hash_length
```